### PR TITLE
Abstract export

### PR DESCRIPTION
**Pull request by jeremyvergnas**
Add multiple export formats support

Also using datatable columns as columns headers for csv export.

### CODE REVIEW

File: src/datatable-export.js [hunk 1]
```diff
@@ -1,38 +1,82 @@
...
+  }
+
+  download(data) {
+    let blob = new Blob([data]);
+
+    if (window.navigator.msSaveOrOpenBlob) {
+      return window.navigator.msSaveBlob(blob, this.filename + '.' + this.format);
+    }
+
+    let a = window.document.createElement('a');
```
Comment (doktordirk) [inline src/datatable-export.js]: use aurelia-pal instead of window here

File: src/datatable-export.js [hunk 2]
```diff
@@ -1,38 +1,82 @@
...
+
+    cleanedColumns = cleanedColumns.replace(/\s/g, '');
+
+    return cleanedColumns;
+  }
+
+  download(data) {
+    let blob = new Blob([data]);
+
+    if (window.navigator.msSaveOrOpenBlob) {
```
Comment (doktordirk) [inline src/datatable-export.js:52]: drop a ie comment

File: src/datatable-export.js [hunk 3]
```diff
@@ -1,38 +1,82 @@
...
-        a.download = 'export.csv';
-        document.body.appendChild(a);
-        a.click();
-        document.body.removeChild(a);
-      } catch (err) {
-        this.datatable.triggerEvent('exception', {on: 'export', error: err});
-      }
+    let rawColumns = (typeof this.columns === 'string' ? this.getColumns().split(',') : this.columns);
+
+    this.datatable.gatherData(this.criteria).then(result => {
```
Comment (doktordirk) [inline src/datatable-export.js]: add return i guess

File: src/datatable-export.js [hunk 4]
```diff
@@ -1,38 +1,82 @@
...
+
+    if (window.navigator.msSaveOrOpenBlob) {
+      return window.navigator.msSaveBlob(blob, this.filename + '.' + this.format);
+    }
+
+    let a = window.document.createElement('a');
+
+    a.href     = window.URL.createObjectURL(blob, {type: 'text/plain'});
+    a.download = this.filename + '.' + this.format;
+    document.body.appendChild(a);
```
Comment (doktordirk) [inline src/datatable-export.js]: same for document

**Comment by jeremyvergnas** [comment]
@doktordirk done

**Comment by jeremyvergnas** [comment]
@doktordirk 👍

**Review by doktordirk — APPROVED** [review]

**Pull request merged**